src/Google/GlobalSiteTag.php

```diff
@@ -602,4 +606,116 @@ private function register_js_for_fast_refresh_dev() {
 			false
 		);
 	}
+
+	/**
+	 * Set user data config when Enhanced Conversions is enabled.
+	 *
+	 * @return string|null
+	 */
+	public function get_enhanced_conversion_tag() {
+		$enhanced_conversions = $this->options->get( OptionsInterface::ADS_ENHANCED_CONVERSIONS_ENABLED );
+
+		if ( ! $enhanced_conversions ) {
+			return;
+		}
+
+		// Retrieve user data from the current session.
+		$customer = WC()->session->get( 'customer' );
+
+		$ec_data = [];
+
+		// Add email address to enhanced conversion data.
+		if ( ! empty( $customer['email'] ) ) {
+			$ec_data['sha256_email_address'] = $this->normalize_and_hash( $customer['email'] );
+		}
+
+		// Add address details if available.
+		if ( ! empty( $customer['first_name'] ) && ! empty( $customer['last_name'] ) && ! empty( $customer['postcode'] ) && ! empty( $customer['country'] ) ) {
+			$ec_data['address'] = [
+				'sha256_first_name' => $this->normalize_and_hash( $customer['first_name'] ),
+				'sha256_last_name'  => $this->normalize_and_hash( $customer['last_name'] ),
+				'postal_code'       => $customer['postcode'],
+				'country'           => $customer['country'],
+			];
+
+			if ( ! empty( $customer['address'] ) ) {
+				$ec_data['address']['street'] = $customer['address'];
+			}
+
+			if ( ! empty( $customer['city'] ) ) {
+				$ec_data['address']['city'] = $customer['city'];
+			}
+
+			if ( ! empty( $customer['state'] ) ) {
+				$ec_data['address']['region'] = $customer['state'];
+			}
+		}
+
+		// Phone number can only be added when emaill and/or address is present.
+		if ( empty( $ec_data ) ) {
+			return;
+		}
+
+		// Add phone number if available, requires country code for correct format.
+		if ( ! empty( $customer['phone'] ) && ! empty( $customer['country'] ) ) {
+			$phone = $this->format_phone_to_international( $customer['phone'], $customer['country'] );
+
+			if ( ! empty( $phone ) ) {
+				$ec_data['sha256_phone_number'] = $this->normalize_and_hash( $phone );
+			}
+		}
+
+		if ( empty( $ec_data ) ) {
+			return;
+		}
+
+		// Return the tag.
+		return sprintf(
+			'gtag("set", "user_data", %s);',
+			wp_json_encode( $ec_data ),
```

Comment:
```suggestion
			wp_json_encode( $ec_data )
```

Remove extra trailing comma